spec/services/person_constructor_spec.rb

```diff
@@ -59,6 +59,26 @@
 
       expect(person.location.city).to eq 'city'
     end
+
+    it "normalizes zip codes before assigning to location" do
+      params = { zip_code: '12345-0001' }
+      expected_params = { zip_code: '12345' }
+      stub_person_finder(expected_params)
+
+      person = PersonConstructor.build(params)
+
+      expect(person.location.zip_code).to eq '12345'
+    end
+
+    it "rejects bad zip codes" do
+      params = { zip_code: '1231' }
+      expected_params = { zip_code: nil }
+      stub_person_finder(expected_params)
```

Comment:
this test requires too much setup to test that the correct params are being passed to Location. This is why I think parameter normalization should be in its own class